plugins/scaffolder/src/components/fields/RepoUrlPicker/utils.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export function serializeFormData(data: {
+  host?: string;
+  owner?: string;
+  repo?: string;
+  organization?: string;
+  workspace?: string;
+  project?: string;
+}) {
+  if (!data.host) {
+    return undefined;
+  }
+
+  const params = new URLSearchParams();
+  if (data.owner) {
+    params.set('owner', data.owner);
+  }
+  if (data.repo) {
+    params.set('repo', data.repo);
+  }
+  if (data.organization) {
+    params.set('organization', data.organization);
+  }
+  if (data.workspace) {
+    params.set('workspace', data.workspace);
+  }
+  if (data.project) {
+    params.set('project', data.project);
+  }
+
+  return `${data.host}?${params.toString()}`;
+}
+
+export function splitFormData(url: string | undefined) {
+  let host = undefined;
+  let owner = undefined;
+  let repo = undefined;
+  let organization = undefined;
+  let workspace = undefined;
+  let project = undefined;
+
+  try {
+    if (url) {
+      const parsed = new URL(`https://${url}`);
+      host = parsed.host;
+      owner = parsed.searchParams.get('owner') || undefined;
+      repo = parsed.searchParams.get('repo') || undefined;
+      organization = parsed.searchParams.get('organization') || undefined;
+      workspace = parsed.searchParams.get('workspace') || undefined;
+      project = parsed.searchParams.get('project') || undefined;
+    }
+  } catch {
+    /* ok */
```

Comment:
really? isn't this an odd behavior to have inside this function, rather than the caller deciding whether to explode or deal with it gracefully?

Reply:
Hmm this was just a pure move from before. Happy to remove the catch block though once theres enough tests surrounding everything. Don't really want to change anything code wise in this refactor if we can help it though as it's already big enough to be honest.

Reply:
aight, note for the future then!